Block/Product.php

```diff
@@ -156,6 +156,15 @@ public function getChildren()
             return $children;
         }
 
+        if ($this->getProduct()->getTypeId() == 'grouped') {
```

Comment:
This should reference the constant rather than be hard-coded:

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/GroupedProduct/Model/Product/Type/Grouped.php#L22